keps/sig-auth/20190806-serviceaccount-tokens.md

```diff
@@ -263,6 +264,35 @@ service account token on behalf of pods running on that node. The
 NodeRestriction admission controller will require that these tokens are pod
 bound.
 
+### Safe Adoption
```

Comment:
@liggitt I remember you having an idea in regards to using metrics.  I think it was something like:

1. When token request is enabled, always use long-lived (say one year) but ephemeral tokens with pods (have them bound to the pod's lifetime)
2. The kubelet will re-issue the tokens continuously (say every hour)
3. Have metrics around pods that use an ephemeral token for longer than an hour (i.e. this tells you what pods are not reloading tokens)

I probably missed something...

Reply:
Thread starts here-ish: https://github.com/kubernetes/kubernetes/issues/70679#issuecomment-575602058

Reply:
Essentially the idea is the same.
1. When token is requested, kube apiserver always give back long-lived token (but tell the caller that this token is expiring soon, like one hour).
2. Project volume would periodically re-issue token request when current one close to expiration (1 hour).
3. Have metric in kube apiserver to report stale token usage.